sdk/storage/Azure.Storage.DataMovement/src/DisabledTransferCheckpointer.cs

```diff
@@ -18,6 +19,7 @@ public override Task AddNewJobAsync(
             StorageResource destination,
             CancellationToken cancellationToken = default)
         {
+            CancellationHelper.ThrowIfCancellationRequested(cancellationToken);
```

Comment:
This file can be reverted. None of these tokens will ever have the opportunity to go off. These are all sync methods that instantly return.

Reply:
Yes I get that that the implementation will instantly return but I feel like if you are passing CancellationToken to a method, you are expecting it to check for the status of the CancellationToken at some point. If you are not, then why pass in a CancellationToken? Is this reasoning wrong?

Reply:
Generally, yes you should either check it or forward it. But that starts to get strange when you're given an interface or abstract base class that defines an async API, but a particular implementation may not end up being sync at all. Various implementations need to be async and use that cancellation token, but this one happens to be purely synchronous and so doesn't end up using it. Were this a long-running sync method we might check it at various points in the loop, but this is not that case.